phaser/phase.py

```diff
@@ -21,8 +21,8 @@ def __init__(self,
         self.context = context or Context()
         self.steps = steps or self.__class__.steps
         self.renumber = renumber
-        self.extra_sources = extra_sources or getattr(self.__class__, 'extra_sources', [])
-        self.extra_outputs = extra_outputs or getattr(self.__class__, 'extra_outputs', [])
+        self.extra_sources = extra_sources or deepcopy(getattr(self.__class__, 'extra_sources', []))
+        self.extra_outputs = extra_outputs or deepcopy(getattr(self.__class__, 'extra_outputs', []))
```

Comment:
It may help a future developer to know why we need to do a `deepcopy` here.  I know you ran into a bug that required this to fix, but I already cannot remember what it was :)